tests/TestCase/ORM/TableTest.php

```diff
@@ -217,6 +221,66 @@ public function testTableQuery(): void
         );
     }
 
+    /**
+     * Tests subquery() disables aliasing.
+     */
+    public function testSubqueryAliasing(): void
+    {
+        $articles = $this->getTableLocator()->get('Articles');
+        $subquery = $articles->subquery();
+
+        $subquery->select('Articles.field1');
+        $this->assertRegExpSql(
+            'SELECT <Articles>.<field1> FROM <articles> <Articles>',
+            $subquery->sql(),
+            !$this->connection->getDriver()->isAutoQuotingEnabled()
+        );
+
+        $subquery->select($articles, true);
+        $this->assertEqualsSql('SELECT id, author_id, title, body, published FROM articles Articles', $subquery->sql());
```

Comment:
Does this also need to handle the situation where auto-quoting is enabled?

Reply:
Hmm it would seem so but there are no test failures so I guess not. It's an existing test case which I just moved from `QueryTest` to `TableTest`.